### PR TITLE
Stream deletion API

### PR DESCRIPTION
**Pull request by Jmgr**
This PR adds an RPC so that clients can ask for the deletion of a stream. All corresponding partitions will also be removed, as well as their persistent storage.

Note that Liftbridge clients that have subscribed to a stream that will get deleted do not receive any notification.

A protobuf generated file has been added in a separate commit since it is suspected that a different version of protoc is being used, so those changes may or may not be kept.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,12 @@ module github.com/liftbridge-io/liftbridge
 
 go 1.13
 
+replace github.com/liftbridge-io/liftbridge-api => ../liftbridge-api
```
Comment (cruickshankpg) [inline go.mod]: these should replace with github.com/ably-forks/liftbridge

File: server/partition.go [hunk 2]
```diff
@@ -370,16 +380,9 @@ func (p *partition) stopLeading() error {
...
-		// Stop leading if previously a leader.
-		if err := p.stopLeading(); err != nil {
-			return err
-		}
-	} else if p.isFollowing {
-		// If previously a follower, we need to reset.
-		if err := p.stopFollowing(); err != nil {
-			return err
-		}
+	err := p.stopLeadingOrFollowing()
```
Comment (cruickshankpg) [inline server/partition.go]: collapse this into an `if err := p.stop(); err != nil {`

Comment (Jmgr) [reply server/partition.go]: :+1:

File: go.mod (hunk 1 above)
Comment (Jmgr) [reply go.mod]: Yes, I'll have to merge the other PR in ably-forks/liftbridge-api before that though.

**Comment by Jmgr** [comment]
Merged into upstream, closing.

**Pull request closed without merging**